Extension/src/LanguageServer/client.ts

```diff
@@ -1551,6 +1553,15 @@ export class DefaultClient implements Client {
                 if (changedSettings["legacyCompilerArgsBehavior"]) {
                     this.configuration.handleConfigurationChange();
                 }
+                if (changedSettings["legacyCompilerArgsBehavior"]) {
+                    this.configuration.handleConfigurationChange();
+                }
+                if (changedSettings["default.compilerPath"]) {
+                    ui.showCompilerStatusIcon(false);
+                }
+                if (changedSettings["default.compileCommands"]) {
```

Comment:
Do we need to add a case for `default.configurationProvider`?  Since the body of the `if` is the same for each changed case, you can probably `||` all the conditions together into a single `if`.

Reply:
`changedSettings["default.compileCommands"]` can also be part of the conditional `changedSettings["default.compilerPath"] || changedSettings["default.configurationProvider"]` because both conditional statements result in doing the same logic when both are true.